### DISCUSSION TITLE
What version of linux and support library can install Genesis-1.3-v4?

### DISCUSSION THREAD

**Discussion by Ykokol** [svenreiche/Genesis-1.3-Version4#91] [Q&A] [answered]
What version of linux and support library can install Genesis-1.3-v4?
I try to install it in ubuntu ,and cmake is successful ,but it Looks like a compilation error when make --" FTTObj is not a name type" .
 i had install the libraries :gcc g++ libfftw3-dev libhdf5-openmpi-dev et.al. my ubuntu version as 22.04 or 20.04. 
it solved，but occur a new error that :when i continue make to step " Linking CXX executable genesis4", it print such as "undefinded H5Pcreat" et. then print make failure cause error.

add one :although cmake successful ,but it note h5pcc not found (my system had h5pcc)
hope to reply! thank you

**Comment by svenreiche** [comment, marked as answer]
Hi,
in principle it should work for you. Myself I have compiled Genesis ver.4 on Ubuntu 20.4 and 22.4 with not much problems.
Myself I am using c++ and cc and not the wrapper h5pcc (but it should work with that as well). Nevertheless it looks for it. If it is found it skips the part of adding a bunch of libraries. In a nutshell it requires openmpi, hdf5 with support for parallel access and the fftw3. Sometimes it is important in which order packages and libraries are installed. E.g. if hdf5 is installed before openmpi it would only generate the single processor verison.

Here is the output of CMAKE command:

/home/reiche/snap/clion-2020.1.2/bin/cmake/linux/bin/cmake -DCMAKE_BUILD_TYPE=Debug -G "CodeBlocks - Unix Makefiles" /data/Codes/Genesis/Version4
-- h5pcc not found - looking for basic compiler
-- C++ Compiler: /usr/bin/c++
-- C Compiler: /usr/bin/cc
-- MPI Found
-- HDF5 Found
-- Version: 1.10.7
-- Parallel HDF5 supported
-- FFTW found
-- Build Info:
-- C++ Compiler: /usr/bin/c++
-- C Compiler: /usr/bin/cc
-- CMAKE_C_FLAGS: 
-- CMAKE_CXX_FLAGS: '-lstdc++'
-- Used libraries: MPI::MPI_C;HDF5::HDF5;PkgConfig::FFTW
-- Defining the macro FFTW in the source code
-- Configuring done
-- Generating done
-- Build files have been written to: /data/Codes/Genesis/Version4/cmake-build-debug

**Reply by Ykokol** [reply]
yeah, my cmake output is same as your's ,maybe i should re-install the libraries in a different order!
thanks for your reply !
best wish for you!

**Reply by svenreiche** [reply]
Hope that it works out. Otherwise the compilation output should help us more.

**Reply by caoyuteng** [reply]
Hi,my cmake output is same as Ykokol，and i have re-install the libraries ,but it still have the same problem.I wonder if openmpi and hdf5 versions are incompatible? Could you please tell us the specific openmpi and hdf5 versions? and provide  a detailed tutorial on installing Genesis in Ubuntu 20.04

**Reply by ken-lauer** [reply]
Genesis 4 is now available on conda-forge with both OpenMPI/MPICH builds, if you just want to use the program. Some basic instructions on how to install and use conda/miniforge from scratch:

<details>

<summary>conda installation details</summary>


> Download and install the latest [[Miniforge](https://conda-forge.org/miniforge/)](https://conda-forge.org/miniforge/) for your platform.
> For Linux on x86-64, download: https://github.com/conda-forge/miniforge/releases/download/24.1.2-0/Mambaforge-24.1.2-0-Linux-x86_64.sh
> - With M1/M2/M3 processors, download: https://github.com/conda-forge/miniforge/releases/download/24.1.2-0/Mambaforge-24.1.2-0-MacOSX-arm64.sh
> For MacOS:
> - With M1/M2/M3 processors, download: https://github.com/conda-forge/miniforge/releases/download/24.1.2-0/Mambaforge-24.1.2-0-MacOSX-arm64.sh
> - For older Intel macs, download the x86-64 version
> 
> This is a customized version of miniconda that includes the fast mamba solver by default and automatically sources packages from `conda-forge`.
> 
> Open a terminal and enable `conda`-related commands by doing the following:
> ```bash
> conda init
> ```
> To use conda commands, you now need to open a new session of your terminal program.
>
> To install the OpenMPI version of Genesis 4, perform the following:
>
> ```bash
> conda create -n genesis4 genesis4=*=mpi_openmpi* 
> ```
>
> To install the MPICH version of Genesis 4, perform the following:
>
> ```bash
> conda create -n genesis4 genesis4=*=mpi_mpich* 
> ```
>
> To use Genesis4 from that environment, first activate the environment:
>
> ```bash
> conda activate genesis4
> genesis4 --help
> ```

</details>

**Reply by caoyuteng** [reply]
Sorry,I can't understand what you mean. Do you mean to install conda-forge first in Ubuntu 20.04 system? Then install openmpi and hdf5? Since I had previously installed Genesis version 4.5.1, which I accidentally destroyed, for  Genesis version 4.5.1, i don't need to install conda-forge,  just need to install openmpi and hdf5,and then make

**Reply by ken-lauer** [reply]
If you'd like to build from source, you can ignore what I'm suggesting entirely (and then I would apologize for taking this thread on a tangent!).

However, if you want an easy method to install Genesis4 **binaries** (including HDF5, OpenMPI, and all required dependencies), I would suggest learning a bit about conda and then taking another look at my message above (or #178).  Conda is a package manager which works on Linux, MacOS, and Windows. The conda-forge channel is where we collaborate to release many scientific packages for the community - at least partially to avoid the headaches that come with manual compilation and managing dependencies on your own.

Here's some more information if you'd like to explore this path: https://conda.io/projects/conda/en/latest/user-guide/getting-started.html

**Comment by ZeugAusHH** [comment]
Dear @caoyuteng ,

I made some documentation how to build "Genesis 1.3" on Ubuntu 22.04LTS, starting from a fresh installation. Please see here: https://github.com/ZeugAusHH/Genesis-1.3-Version4/tree/cl_20240625__builddoc.ubuntu/manual/build_notes/ubuntu_22.04

There, I documented the output of CMake (and also a list of packages installed on the system).

Hope this helps to solve your issue.

Best
Christoph
